File: python/InsertionSort.py
class InsertionSort:
    def insert(arr):
        for i in range(len(arr)-1):
            for j in range(i+1, 0, -1):
                if (arr[j-1] > arr[j]):
                    arr[j-1], arr[j] = arr[j], arr[j-1]
                else:
                    break


arr = [21, 32, 1, 43, 0]
InsertionSort.insert(arr)
print(arr)

File: Java/InsertionSort.java
package Java;

import java.util.Arrays;

public class InsertionSort {
    public static void main(String[] args) {
        int[] arr = { 32, 42, 1, 4, 2, 5 };
        insert(arr);
        System.out.println(Arrays.toString(arr));

    }

    public static void insert(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = i + 1; j > 0; j--) {
                if (arr[j] < arr[j - 1]) {
                    swap(arr, j, j - 1);
                } else {
                    break;
                }
            }
        }

    }

    static void swap(int[] arr, int start, int end) {
        int temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
    }

}
